### DISCUSSION TITLE
Announcement: JorjMcKie is a danged psychic (thank you!)

### DISCUSSION THREAD

**Discussion by mlissner** [pymupdf/PyMuPDF#1352] [Announcements]
In my project, [xray](https://free.law/projects/x-ray/), we've been trying to figure out what to do about mixed colorspaces between text and rectangles. 

We want to detect if text and drawings are the same color, but it was hard since they could be in different colorspaces. We discussed it as a team internally, decided figuring out colorspaces was really hard, and gave up.

Enter PyMuPDF 1.19.1, and colorspaces are all normalized to RGB. Incredible! 

We just simplified our code:

https://github.com/freelawproject/x-ray/commit/5c3657e6b2853d41414cb61780390d359daad636

And this fixes an issue we had given up on! 

Pretty wild. @JorjMcKie, are you reading our code?

**Comment by JorjMcKie** [comment]
😂😂😂
> @JorjMcKie, are you reading our code?

I swear I did not until right now. And thank you very much for this big compliment - so prominently positioned.

However, as a sober naturalist, I - reluctantly in this case - have to say that a seemingly paranormal situation is just one where not all facts are known.
This is the case here, too. Sheer coincidence: A user was trying to repaint extracted drawings on some new page and came across an exotic colorspace with just two color components, which PyMuPDF (with its restriction to the basic set GRAY, RGB, CMYK) refused to accept.

This kicked me off digging for a solution. I was not aware of your similar issue.

**Reply by mlissner** [reply]
I saw that other conversation, but a part of me wondered if you saw our issue too and said, "Fine, two people need this, I'll do it." Anyway, this is great, thank you again.
